eclipse-glsp/glsp-playwright
# Update project and implement select tests

## Description

This PR brings some necessary changes for the future:

1) Updates the used frameworks and introduces fixes to work with them:
- Theia-Playwright
- GLSP-Client
- Playwright
2) Restructures example tests to be more align with `GLSP-Client` project structure
3) Make `GLSP-Playwright` more defensive (e.g., more information, checks, vscode-setup)

Finally, test cases based for [select](https://github.com/eclipse-glsp/glsp-client/tree/master/packages/client/src/features/select) are implemented.

#### What it does

- Update folder structure to be similar to glsp-client
- Remove aliasing from example project
- Fix failing tests due to ghost elements

#### New Features

- [skipIntegration](https://github.com/eclipse-glsp/glsp-playwright/pull/7/files#diff-1e9fe821e0ed3f86d5c21a5d3f9b1a56b4dd8fddefd6f49579bb56a9b561728dR129) to skip specific integrations
- Introduce `*` type for models to allow for all types
- Make access more defensive

#### How to test
 
Please prepare the project by reading the [README](https://github.com/eclipse-glsp/glsp-playwright/tree/main/examples/workflow-test#eclipse-glsp-playwright-example)
 
1) Run the tests as described there.

The `standalone` version succeeds all the thests.

Theia and VSCode fail the test:

```
    [...] › ../../tests/features/select/select.spec.ts:84:9 › The select feature › should allow to deselect a single element through a keybinding 
```

It seems that using `Escape` does not reset the `select` state.

#### Follow-ups

-

#### Changelog


##### New way of creation

- GLSPApp

`GLSP-Playwright` follows a similar approach as `Theia-Playwright`. However, we do not require any `loader`. That means we can create our `*App` now directly.

Before
```
app = await GLSPApp.loadApp(WorkflowApp, {
    type: 'integration',
    integration
});
```

Now
```
app = new WorkflowApp({
    type: 'integration',
    integration
});
```

- All integrations take now a `IntegrationArgs` as first constructor parameter
